### ISSUE TITLE
Update WITHOUT losing data?

### ISSUE THREAD

**Issue by mitchellurgero** [mailcow/mailcow-dockerized#1277]
I just tried to update and without warning ALL my data was deleted.
If the data needs to be purged, there should be an extra warning saying this. 

Documentation did not have any warnings on this either.

Is there a way to update without losing data? Luckily I only had a couple days emails.

**Comment by mritzmann** [comment]
https://github.com/mailcow/mailcow-dockerized/issues/1178

**Comment by mitchellurgero** [comment]
Sorry, I am very new to docker, I read the issue, all comments, etc. Can you summarize the steps needed to get old data (If they still exist)?

**Comment by andryyy** [comment]
Would be interesting to know what your current COMPOSE_PROJECT_NAME is set to in mailcow.conf right now. 

No data is lost anyway. What’s the output of “docker volume ls”?

**Comment by mitchellurgero** [comment]
COMPOSE_PROJECT_NAME=mailcowdockerized

and that command shows all volumes:

mailcowdockerized_*-vol-1
and
mailcow-dockerized_*-vol-1

EDIT: I changed the COMPOSE_PROJECT_NAME to mailcow-dockerized and everything came back, thanks for the hint! I legit had a heart attack!

**Comment by mitchellurgero** [comment]
Would changing my compose name in the conf and "rebooting" of the docker containers help fix this?

**Comment by mkuron** [comment]
Yes, but the update.sh script may automatically change it back in the future. So I would recommend deleting the (almost) empty set of folders and rename the others appropriately.

**Comment by mitchellurgero** [comment]
I would rather not as I have DKIM setup and such, would it hurt to just change that variable in the future or should I restart from scratch?

**Comment by mitchellurgero** [comment]
Also why would that change??

EDIT: If I keep it as "mailcowdockerized" will I be safe with updates?

**Comment by andryyy** [comment]
You probably didn’t stop the stack before fixing the COMPOSE_PROJECT_NAME variable?

Please read the other issue... you will find information about why it changed.

**Comment by andryyy** [comment]
Oh forget it. Didn’t see the edit.

**Comment by mitchellurgero** [comment]
Thanks for the information! I see now. So If I keep the new compose name I will be good for updates? :)

**Comment by mitchellurgero** [comment]
Closing as I was able to resolve this with help! Thanks!
